runelite-client/src/main/java/net/runelite/client/plugins/inferno/InfernoWaveMappings.java

```diff
@@ -26,121 +26,138 @@
+		waveMapBuilder.put(43, new int[]{32, 32, 32, 85, 240, 490});
+		waveMapBuilder.put(44, new int[]{32, 32, 32, 85, 85, 240, 490});
+		waveMapBuilder.put(45, new int[]{32, 32, 32, 165, 240, 490 });
+		waveMapBuilder.put(46, new int[]{32, 32, 32, 85, 165, 240, 490});
+		waveMapBuilder.put(47, new int[]{32, 32, 32, 85, 85, 165, 240, 490});
+		waveMapBuilder.put(48, new int[]{32, 32, 32, 165, 165, 240, 490});
+		waveMapBuilder.put(49, new int[]{32, 32, 32, 240, 240, 490});
+		waveMapBuilder.put(50, new int[]{32, 32, 32, 370, 490});
+		waveMapBuilder.put(51, new int[]{32, 32, 32, 85, 370, 490});
+		waveMapBuilder.put(52, new int[]{32, 32, 32, 85, 85, 370, 490});
+		waveMapBuilder.put(53, new int[]{32, 32, 32, 165, 370, 490});
+		waveMapBuilder.put(54, new int[]{32, 32, 32, 85, 165, 370, 490});
+		waveMapBuilder.put(55, new int[]{32, 32, 32, 85, 85, 165, 370, 490});
+		waveMapBuilder.put(56, new int[]{32, 32, 32, 165, 165, 370, 490});
+		waveMapBuilder.put(57, new int[]{32, 32, 32, 240, 370, 490});
+		waveMapBuilder.put(58, new int[]{32, 32, 32, 85, 240, 370, 490});
+		waveMapBuilder.put(59, new int[]{32, 32, 32, 85, 85, 240, 370, 490});
+		waveMapBuilder.put(60, new int[]{32, 32, 32, 165, 240, 370, 490});
+		waveMapBuilder.put(61, new int[]{32, 32, 32, 85, 165, 240, 370, 490});
+		waveMapBuilder.put(62, new int[]{32, 32, 32, 85, 85, 165, 240, 370, 490});
+		waveMapBuilder.put(63, new int[]{32, 32, 32, 165, 165, 240, 370, 490});
+		waveMapBuilder.put(64, new int[]{32, 32, 32, 85, 240, 240, 370, 490});
+		waveMapBuilder.put(65, new int[]{32, 32, 32, 85, 370, 370, 490});
+		waveMapBuilder.put(66, new int[]{32, 32, 32, 85, 490, 490});
+		waveMapBuilder.put(67, new int[]{900});
+		waveMapBuilder.put(68, new int[]{900, 900, 900});
+		waveMapBuilder.put(69, new int[]{1400});
+
+		waveMapping = waveMapBuilder.build();
+
+		ImmutableMap.Builder<Integer, String> nameMapBuilder = new ImmutableMap.Builder<>();
+
+		nameMapBuilder.put(32, "Jal-Nib - Level 32");
+		nameMapBuilder.put(85, " Jal-MejRah - Level 85");
+		nameMapBuilder.put(165, "Jal-Ak - Level 165");
+		nameMapBuilder.put(240, "Jal-ImKot - Level 240");
+		nameMapBuilder.put(370, "Jal-Xil - Level 370");
+		nameMapBuilder.put(490, "Jal-Zek - Level 490");
+		nameMapBuilder.put(900, "JalTok-Jad - Level 900");
+		nameMapBuilder.put(1400, "TzKal-Zuk - Level 1400");
+
+		npcNameMapping = nameMapBuilder.build();
+	}
 
-	static Map<Integer, String> npcNameMapping()
+	static void addWaveComponent(PanelComponent panelComponent, String header, int wave, Color titleColor, Color color)
+	{
+		int[] monsters = waveMapping.get(wave);
+
+		panelComponent.getChildren()
+			.add(TitleComponent.builder()
+				.text(header)
+				.color(titleColor)
+				.build()
+			);
+
+
+		for (int i = 1; i < monsters.length; i++)
 		{
-			return new HashMap<Integer, String>()
+			int monsterType = monsters[i];
+			int count = 1;
+
+			for (; i < monsters.length - 1 && monsters[i + 1] != monsterType; i++)
```

Comment:
should be for (; i < monsters.length - 1 && monsters[i + 1] == monsterType; i++)